### PR TITLE
Compatibility with MySQL Group Replication and MySQL InnoDB Cluster

### LINKED ISSUE #11596: Compatibility of Piwik with MySQL 5.7.17 in group replication
Reported by email:

When MySQL 5.7.17 is running in group replication (a feature introduced in MySQL 5.7.17), all tables are required to have a primary key. Installing Piwik 3.0.3 on MySQL 5.7.17 running in a group replication settings, the replication breaks the group because all of the Piwik tables do not contain a primary key. The solution is to add a primary key to the three Piwik tables that do not contain primary keys:

piwik_log_profiling
piwik_plugin_setting
piwik_site_setting

### PR DESCRIPTION
**Pull request by ghost**
Columns added to support MySQL group replication (every table must have a primary key)

### CODE REVIEW

**Comment by sgiehl** [comment]
Besides changing the schema, there also needs to be an update script, that performs the changes while updating.
Also I'm not sure if `INTEGER` is the right choice. We are already having issues at other places where `INTEGER` is simply too small.

**Comment by Aardvark8** [comment]
Integer was changed to BIGINT.  Where would we place the update script.  Which file is best to add the update script?

**Comment by sgiehl** [comment]
See https://github.com/piwik/piwik/tree/3.x-dev/core/Updates
As I don't think this will get merged for 3.0.5, maybe you could name the Update-Script 3.1.0 already.

**Comment by Aardvark8** [comment]
Update script was created core/Updates/3.1.0.php.  Hopefully we could be considered for 3.1.0.

**Comment by Aardvark8** [comment]
Tested the update script and it works well.  Retested and everything checks out and we are ready for merge.

File: core/Db/Schema/Mysql.php [hunk 1]
```diff
@@ -81,6 +81,8 @@ public function getTablesCreateSql()
                           `setting_name` VARCHAR(255) NOT NULL,
                           `setting_value` LONGTEXT NOT NULL,
                           `user_login` VARCHAR(100) NOT NULL DEFAULT '',
+						  `plugin_setting_id` BIGINT UNSIGNED NOT NULL AUTO_INCREMENT,
```
Comment (mattab) [inline core/Db/Schema/Mysql.php]: Would it be possible to make the spacing consistent with the lines above (ie. use spaces not tabs)?

**Review by mattab — CHANGES_REQUESTED** [review]

File: core/Db/Schema/Mysql.php (hunk 1 above)
Comment (Aardvark8) [reply core/Db/Schema/Mysql.php]: Done, tabs were changed to spaces in Mysql.php.

**Comment by Aardvark8** [comment]
Ok, tabs were changed to spaces.

File: core/Updates/3.2.0.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        $updater->executeMigrations(__FILE__, $this->getMigrations($updater));
+    }
+
+    /**
+     * @param Migration[] $queries
+     * @return Migration[]
+     */
+    private function getPluginSettingsMigrations($queries)
+    {
+		$queries[] = $this->migration->db->addColumn($this->pluginSettingsTable, 'plugin_setting_id', 'BIGINT UNSIGNED NOT NULL PRIMARY KEY AUTO_INCREMENT', 'user_login');
```
Comment (mattab) [inline core/Updates/3.2.0.php]: Hi @Aardvark8 fyi you missed a few tabs here

**Review by ghost — COMMENTED** [review]
ok, tabs were replaced with spaced in 3.2.0.php as well.

**Comment by mattab** [comment]
Hi @Aardvark8 

When trying to run the upgrade it fails because there is already one field with auto increment in `site_setting` table, see:
```
    The database upgrade process may take a while, so please be patient.

  Executing ALTER TABLE `piwik_plugin_setting` ADD COLUMN `idplugin_setting` BIGINT UNSIGNED NOT NULL PRIMARY KEY AUTO_INCREMENT;... Done. [1 / 3]
  Executing ALTER TABLE `piwik_site_setting` ADD COLUMN `idsite_setting` BIGINT UNSIGNED NOT NULL PRIMARY KEY AUTO_INCREMENT;... 
    [X] Critical Error during the update process:

    * core/Updates/3.2.1.php:
    Error trying to execute the migration 'ALTER TABLE `piwik_site_setting` ADD COLUMN `idsite_setting` BIGINT UNSIGNED NOT NULL PRIMARY KEY AUTO_INCREMENT;'.
```

**Comment by Aardvark8** [comment]
Hi Matt, We do not receive that error.  Looking at the file, there does not appear to be any other column in that table with AUTO_INCREMENT.  Is it possible, that your tester ran the code twice and received the error on the second run (maybe after adding the primary key column by the original name)?

**Comment by diosmosis** [comment]
@mattab merging since it works for me & build passes.

Thanks for submitting your first PR @Aardvark8 !

**Comment by mattab** [comment]
On demo2 we're getting the error
```
    Matomo database will be upgraded from version 3.5.1 to the new version 3.6.0-b1.

    The database upgrade process may take a while, so please be patient.

  Executing ALTER TABLE `plugin_setting` ADD COLUMN `idplugin_setting` BIGINT UNSIGNED NOT NULL PRIMARY KEY AUTO_INCREMENT;... Done. [1 / 3]
  Executing ALTER TABLE `site_setting` ADD COLUMN `idsite_setting` BIGINT UNSIGNED NOT NULL PRIMARY KEY AUTO_INCREMENT;... 
    [X] Critical Error during the update process:

    * core/Updates/3.6.0-b1.php:
    Error trying to execute the migration 'ALTER TABLE `site_setting` ADD COLUMN `idsite_setting` BIGINT UNSIGNED NOT NULL PRIMARY KEY AUTO_INCREMENT;'.
    The error was: SQLSTATE[42000]: Syntax error or access violation: 1075 Incorrect table definition; there can be only one auto column and it must be defined as a key


```

so this is not working as expected? cc @diosmosis

**Comment by diosmosis** [comment]
Looks like this problem should have been fixed by `https://github.com/matomo-org/matomo/blob/3.x-dev/core/Updates/3.0.0-b1.php#L172-L177`, not sure why this did not get applied on demo2.

**Pull request merged**